src/persist/src/indexed/columnar.rs

```diff
@@ -438,4 +564,31 @@ mod tests {
             .collect();
         assert_eq!(reads, updates);
     }
+
+    #[test]
+    fn vec_builder() {
+        fn testcase(max_len: usize, key_size: usize, num_records: usize) {
+            let key = "x".repeat(key_size);
+            let expected = (0..num_records)
+                .map(|x| ((key.as_bytes(), "".as_bytes()), u64::cast_from(x), 1))
```

Comment:
minor nit that this test doesn't exercise the values overflowing the size limit but it seems like it easily could

Reply:
done. also added coverage of the number of ColumnarRecords each case is expected to result in